website/docs/externaldata.md

```diff
@@ -0,0 +1,154 @@
+---
+id: externaldata
+title: External Data
+---
+
+> This feature is still in alpha stage, so the final form can still change (feedback is welcome!). Mutation is not yet supported with external data.
+
+## Motivation
+
+Gatekeeper provides various means to mutate and validate Kubernetes resources. However, in many of these scenarios this data is either built-in, static or user-defined. With external data feature, we are enabling Gatekeeper to interface with various external data sources, such as image registries, using a provider-based model.
+
+A similar way to connect with an external data source can be done today using OPA's built-in `http.send` functionality. However, there are limitations to this approach.
+- Gatekeeper does not support rego policies for mutation, which can not use the OPA http.send built-in function
+- There are unknown factors when the proxy endpoint is not responsive, doesn't exist/down or fails. These are addressed by external data interfaces such as timeout.
+
+Using the external data solution provides benefits, such as:
+- Tackle common scenarios together, such as image SHA mutation that many of these scenarios will utilize.
+- Provider based model provides examples and guidelines for each provider to integrate with Gatekeeper
```

Comment:
Provider model creates a common interface for extending Gatekeeper with external data.

This means that developers and consumers of data sources can rely on that common protocol to ease authoring of both constraint templates and data sources. It also makes change management a bit easier... if a data source is compliant with external data, users of that source should be able to tell whether upgrading it will break existing CTs (and our goal is to have that answer always be "no").

It also allows us to provide "free" value as a platform, like the performance benefits for mutation.

Reply:
Thank you! updated.